### ISSUE TITLE
Quotes not working 

### ISSUE THREAD

**Issue by rahulshah9713** [Kotak-Neo/kotak-neo-api#157]
client.quotes gives `None` as output

```
instrument_tokens = [{"instrument_token": "21951", "exchange_segment": "nse_cm"}]
print(client.quotes(instrument_tokens, isIndex=False, quote_type="ltp"))
```

**Comment by nishantmoreatgit** [comment]
Yes facing difficulty in fetching client.quotes...giving None output

**Comment by rahulshah9713** [comment]
@Kotak-Neo when will this be resolved?

**Comment by ronak24069** [comment]
You can use the previous version, this is a bug from the current version

**Comment by rahulshah9713** [comment]
> You can use the previous version, this is a bug from the current version

How can that be done? Can you please give me pip command for the same?

**Comment by ronak24069** [comment]
You cannot do that using pip now, what you can do is replace the current 'neo_api_client' package folder with the previous version folder. If you don't have one, let me know, will share the zip through mail.

**Comment by rahulshah9713** [comment]
> You cannot do that using pip now, what you can do is replace the current 'neo_api_client' package folder with the previous version folder. If you don't have one, let me know, will share the zip through mail.

Yes, please share here itself as attachment so others can use it as well

**Comment by arunjain9988** [comment]
@Kotak-Neo,  can we check and resolve this.

**Comment by ronak24069** [comment]
Sure, please check this out.

[neo_api_client.zip](https://github.com/Kotak-Neo/kotak-neo-api/files/14900704/neo_api_client.zip)

**Comment by deepakbairy** [comment]
Yes, this is working for me

**Comment by mshh9** [comment]
Last year they completely revamped the API and made it to NEO. And now with version 1.0.1 updates things dont work the same way starting with session init. @Kotak-Neo if you are making changes please ensure that it works

**Comment by dev3180** [comment]
dear its new version

**Comment by dev3180** [comment]
lets me check, wait

**Comment by dev3180** [comment]
its modified version, u can use it

**Comment by ronak24069** [comment]
> i got solution for quote, everything work perfectly [neo_api_client.zip](https://github.com/Kotak-Neo/kotak-neo-api/files/14979792/neo_api_client.zip)

You updated the script yourself?

**Comment by dev3180** [comment]
yes

**Comment by dev3180** [comment]
u can check its new version

**Comment by dev3180** [comment]
sure, u can, its modified

**Comment by ronak24069** [comment]
Please raise a pull request then for the same, let's see if they merge it

**Comment by dev3180** [comment]
sure dear

**Comment by deepakbairy** [comment]
I am trying to get an option price. I see the newer fix works sometimes, but sometimes gives empty response.
Hello repo owners. Please make sure to test this quotes API before creating pull request.

**Comment by dev3180** [comment]
deepak shre ur code here, will provide solution if possible

**Comment by dev3180** [comment]
its working without any error, i tested it

**Comment by dev3180** [comment]
ye kotak wale pata nahi yaar kya karte rehte hai,   

yaar aap log sahi code chalne doge ya nahi ?????????????

ya fir kotak se account delete kar de ??????   aap logo ki service bhi bahot bakwas hai, 1 week pehle trade api ki team se help mangi hai, ab talak nahi mili, 

aap logo ki service bhi bahot bekar hai, aapke web module, mobile app and trading api  sach me bahot bakwas hai,

**Comment by dev3180** [comment]
even ohlc data bhi provide nahi karte ye log

**Comment by dev3180** [comment]
quotes doesn't working in both version, new & old

**Comment by gmarwaha77** [comment]
Hello, 

I was using the earlier KST API and it worked fine, in that the calls to quote were synchronous, with this new Neo API the call to get_quote and subscribe are through websockets running in a thread, I do not need to monitor the price actively it is a good feature but is there a way to get the LTP and market depth at that instant rather than a subscription to the price?

Will the modified file solve this issue can anyone please comment? Kotak is just closing the service requests without responding.

Thanks a lot

**Comment by gmarwaha77** [comment]
> its working without any error, i tested it

it worked thanks a lot I ported your changes to neo_api and NeoWebSocket to get the quote

**Comment by soumyajyotibiswas** [comment]
Hello,

I wrote some sample code, nothing fancy, just of the examples to test how client.quotes would work. But it's giving me None as output. How to resolve this? I tried the zip files in the thread but did not work for me.

**Comment by nishantmoreatgit** [comment]
@Kotak-Neo  why don't you shut your API shop, you don't have courtesy to inform API users about your so called bug which could halt the users code

**Comment by nishu88** [comment]
@dev3180 I don't see your zip file anymore, can you reshare.
Also which files did you change??

**Comment by dev3180** [comment]
yes i will soon nishu

**Comment by dev3180** [comment]
modified api[
[Uploading neo_api_client.zip…]()
](url)

**Comment by dev3180** [comment]
angel one is best for algo trading, leave kotak and join angel one, everytime got error in quotes and many more, but angel one wow amazing, no one error, 

kotak west of time and money

**Comment by vzanvar-wdc** [comment]
It is such a shame that @Kotak-Neo never interacts in the forum.  They make only big claims [here:](https://documenter.getpostman.com/view/21534797/UzBnqmpD)

> _Dependable support
> You can count on our dedicated support team for a quick turnaround for all your queries_

**Comment by vzanvar-wdc** [comment]
> modified api[ Uploading neo_api_client.zip… ](url)

This link does not point to api_client?

**Comment by dev3180** [comment]
quotes can't fetch

kotak neo west of time and money, angel one much batter for algo trading

**Comment by soumyajyotibiswas** [comment]
> quotes can't fetch
> 
> kotak neo west of time and money, angel one much batter for algo trading
> 

I switched to 5paisa. Hope kotak's team fixes all these bumps sooner than later.

**Comment by vzanvar-wdc** [comment]
> quotes can't fetch
> 
> kotak neo west of time and money, angel one much batter for algo trading

+1

**Comment by CREATER247** [comment]
brother I have used your neo_client but it is giving black message and None object at the same time....
Am I doing something wrong. Please help...........

**Comment by vijaykz** [comment]
> can anyone share the sample code for correctly using the callback function with client.quotes?

I have spent several days doing just that.  For @Kotak-Neo, there is no "correct" method.  😜  I got the idea to move to Angel from one of the comments above.  And Angel is already up and running!

**Comment by sudhanshupati** [comment]
The quote method return None but the response is inside on_message method, if you put a print statement inside on_message , you will see the response printed , but you will not get any response from quote method

**Comment by vijaykz** [comment]
> The quote method return None but the response is inside on_message method, if you put a print statement inside on_message , you will see the response printed , but you will not get any response from quote method

This is an incorrect design, so it is difficult to use.

**Comment by sudhanshupati** [comment]
Agreed,, i have stopped using the quote method

**Comment by dev3180** [comment]
kotak neo west of time and money, angel one much batter for algo trading

**Comment by dev3180** [comment]
always problem in quotes, in a week 2 day got response, mostly west of time and money,

even kotak team never give support, leave kotak and join angelone

**Comment by vijaykz** [comment]
I do not think and neither did I bother trying to make it work.  Move on to another broker.
